bindings/twitter/twitter.go

```diff
@@ -58,36 +54,34 @@ func (t *twitterInput) parseMetadata(metadata bindings.Metadata) error {
 	if !f || as == "" {
 		return fmt.Errorf("accessSecret not set")
 	}
+
+	// query set only on input binding
+	// in output binding scenario the query is provided on create
 	q, f := metadata.Properties["query"]
-	if !f || q == "" {
-		return fmt.Errorf("query not set")
+	if !f && q != "" {
```

Comment:
shouldnt this be `if f &&`?

Reply:
good catch, Fixed. 

```go
if f {
  t.query = q
}
```

Given that `t.query` is already a string so even if we reset it to "" it won't matter.